### DISCUSSION TITLE
Wonder if you have encountered poetry semver?

### DISCUSSION THREAD

**Discussion by Greedquest** [python-semver/python-semver#360] [Q&A] [answered]
The poetry package manager has a semver implementation `poetry.core.semver`:

https://github.com/python-poetry/poetry-core/tree/master/src/poetry/core/semver

There is also a standalone package called poetry semver (although it is no longer maintained):

https://github.com/python-poetry/semver

I was just wondering if you've encountered either of those, and if so, how does this package differ from those two?

**Comment by tomschr** [comment, marked as answer]
Thank you @Greedquest for your input.

I haven't encountered this part of poetry. It's the first time. :wink: Looks interesting, thanks for the pointer!

> I was just wondering if you've encountered either of those, and if so, how does this package differ from those two?

Well, from a first glance, the `poetry.core.semver.Version` class is based on dataclasses whereas semver is not. Furthermore, semver currently doesn't support version ranges which the other lib does support (we have an open issue for that #241). **However, the most important difference is that `poetry.core.semver.Version` is not maintained anymore.**

The design and naming are a bit different between both, but from a basic idea they work the same. I assume, when dataclasses were introduced, semver already existed.
